src/go/rpk/pkg/cli/cluster/config/edit.go

```diff
@@ -106,8 +107,13 @@ func executeEdit(
 			editor = fallbackEditor
 		}
 	}
+	// Fix: Issue #17386
+	// Split the editor environment variable to see if it has editor options e.g. code -w, where editor is "code" and its options is "-w"
+	eArgs := strings.Fields(editor)
```

Comment:
This would fail if something was quoted like `foo 'why would I do this'`. We have an indirect dependency on https://github.com/kballard/go-shellquote, maybe we should make that direct and use `shellquote.Split`? Thoughts @r-vasquez?

Reply:
@rockwotj  thanks for reviewing, I see the `strings.Fields` is still able to handle that. Here is the little test I wrote 

```go
package main

import (
	"fmt"
	"os"
	"strings"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestEditorAndOptions(t *testing.T) {

	testCases := map[string]struct {
		editor   string
		fileName string
		want     string
	}{
		"default": {
			editor:   "code -w",
			fileName: "/tmp/one.yaml",
			want:     "code -w /tmp/one.yaml",
		},
		"noOptions": {
			editor:   "code",
			fileName: "/tmp/one.yaml",
			want:     "code /tmp/one.yaml",
		},
		"withQuotedOption": {
			editor:   "foo 'why would I do this'",
			fileName: "/tmp/one.yaml",
			want:     "foo 'why would I do this' /tmp/one.yaml",
		},
	}

	for name, tc := range testCases {
		t.Run(name, func(t *testing.T) {
			//just to mimick to get from environment variable
			os.Setenv("EDITOR", tc.editor)
			editor := os.Getenv("EDITOR")

			eArgs := strings.Fields(editor)
			editor = eArgs[0]
			eArgs = append(eArgs[1:], tc.fileName)
			got := fmt.Sprintf("%s %s", editor, strings.Join(eArgs, " "))
			assert.Equalf(t, tc.want, got, "want %s but got %s", tc.want, got)
		})
	}

}
```

Does your case covered ?

Reply:
Just another thought, if we could make this editor + options as utility function then we can reuse the same logic in all places where we might need to open an editor and makes the logic testable as well. I don't know how to test `child.Run()`

Reply:
If you make `want` a `[]string` I think you'll see what I mean. The quoting won't be preserved when invoking the underlying method.

Here's what the test cases would be:

```
package main

import (
	"fmt"
	"os"
	"strings"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestEditorAndOptions(t *testing.T) {

	testCases := map[string]struct {
		editor   string
		fileName string
		want     []string
	}{
		"default": {
			editor:   "code -w",
			fileName: "/tmp/one.yaml",
			want:     []string{"code", "-w", "/tmp/one.yaml"},
		},
		"noOptions": {
			editor:   "code",
			fileName: "/tmp/one.yaml",
			want:     []string{"code", "/tmp/one.yaml"},
		},
		"withQuotedOption": {
			editor:   "foo 'why would I do this'",
			fileName: "/tmp/one.yaml",
			want:     []string{"foo", "'why would I do this'", "/tmp/one.yaml"},
		},
	}

	for name, tc := range testCases {
		t.Run(name, func(t *testing.T) {
			got := strings.Fields(tc.editor)
			assert.Equalf(t, tc.want, got, "want %s but got %s", tc.want, got)
		})
	}

}
```

Reply:
https://go.dev/play/p/10nfY1fybF0

Reply:
got your point @rockwotj , well I can use the other module you told if thats go with you all.

Reply:
SGTM